src/overview/components/Tooltip.jsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import classNames from 'classnames'
+
+import localStyles from './Tooltip.css'
+import { tooltip } from './tooltips'
+
+const mainTooltipContainer = showTooltip =>
+    classNames({
+        [localStyles.mainTooltip]: true,
+        [localStyles.isActive]: showTooltip,
+    })
+
+const tooltipButton = showTooltip =>
+    classNames({
+        [localStyles.tooltipButtonIcon]: showTooltip,
+    })
+
+const Tooltip = ({
+    showTooltip,
+    toggleShowTooltip,
+    tooltipIndex,
+    onClickRefreshTooltip,
+}) => {
+    const tooltipObject = tooltip(tooltipIndex)
```

Comment:
this is better than before, but the selected tooltip is still not part of the UI state; only the index is, but that is highly coupled to the current data structure. 

If index state is removed and replaced with a tooltip state, then this call can be done in a thunk which dispatches say a `setTooltip` action after calling a `fetchNextTooltip()` function which can be exported from that `tooltips.js` module. 

The `fetchNextTooltip()` fn may be able to keep it's own index state in a closure, separate of redux, and just increment that each time it's called, so each call returns a new tooltip. Let me know if you need any pointers and I can send you a snippet

Reply:
I think we can do easily by just adding one more state `tooltip` in the reducer a d just set with the `incTooltipIndex` and we can pass directly this value to the Tooltip and can use.
@poltak what you think?

Reply:
The most important thing to note is: the index state is highly coupled to your current data source; it shouldn't be part of UI state. 

Say we return to this later and want to update the tooltips data source to come from a DB somewhere. The `tooltipIndex` state then has no meaning and we need to rewrite the UI state. As that index state is highly coupled, keep it with the tooltips data structure. Just expose an impure function that returns a different tooltip every time it's called. Then this function becomes the data source, and it is the only thing we need to change in the future. 

The default `tooltip` state should be `null` or `{}`, and will need to be set sometime (either when user chooses to show the tooltips, or it needs to be hydrated from local storage, if already shown).

You may also realise from this that the `showTooltip` state doesn't really need to be there either; you can simply derive that value based on whether `tooltip` state is set or not. 

Really important to try and get the most-simple state possible as it cascades throughout the rest of the UI and may make things more complicated than they need to be. And you can always use selectors to efficiently perform calcs and derive new related states from those minimal state data.